node/pkg/watchers/solana/tx_for_addr.go

```diff
@@ -109,7 +116,15 @@ func (s *SolanaWatcher) getTransactionSignatures() ([]*rpc.TransactionSignature,
 	numSignatures := MaxSignaturesPerQuery
 	currSig := solana.Signature{}
 
+	startTime := time.Now()
 	for numSignatures == MaxSignaturesPerQuery {
+		if time.Since(startTime) > time.Minute {
+			// It shouldn't take this long to query transactions. Return an error and kill the watcher. We can't
+			// attempt to process the transactions because we read them from most recent backwards, so there could
+			// be a gap and we might miss older transactions.
+			return results, fmt.Errorf("getTransactionSignatures has taken too long, numResults: %d", len(results))
```

Comment:
just noting that this behavior should probably be noted in the release notes and someone should be monitoring for these errors once Fogo is live in mainnet.